core/bin/zksync_api/src/bin/dev-liquidity-token-watcher.rs

```diff
@@ -94,28 +94,51 @@ fn load_tokens(path: impl AsRef<Path>) -> Vec<(String, String)> {
     tokens
 }
 
-#[derive(Debug, Serialize, Deserialize)]
-struct GrqaphqlQuery {
-    query: String,
+async fn handle_get_asset_platforms(storage: web::Data<CoinGeckoStorage>) -> Result<HttpResponse> {
+    let keys: Vec<PlatformId> = storage.keys().map(|key| key.clone()).collect();
+    if keys.len().gt(&(0 as usize)) {
+        return Ok(HttpResponse::Ok().json(vec![AssetPlatform {
+            id: keys[0].clone(),
+            chain_identifier: Some(9999),
+            name: format!("Local Dev"),
+            shortname: format!("localdev"),
+        }]));
+    }
+
+    HttpResponse::InternalServerError().message_body(AnyBody::from_message("Nothing in storage"))
 }
 
-async fn handle_graphql(
-    params: web::Json<GrqaphqlQuery>,
-    volume_storage: web::Data<VolumeStorage>,
+async fn handle_get_coin_contract(
+    path: web::Path<(String, String)>,
+    storage: web::Data<CoinGeckoStorage>,
 ) -> Result<HttpResponse> {
-    // TODO https://linear.app/matterlabs/issue/ZKS-413/support-full-version-of-graphql-for-tokenvalidator
-    let query_parser = Regex::new(r#"\{token\(id:\s"(?P<address>.*?)"\).*"#).expect("Right regexp");
-    let caps = query_parser.captures(&params.query).unwrap();
-    let address = &caps["address"].to_ascii_lowercase();
-    let volume = volume_storage.get_volume(address);
-    let response = GraphqlResponse {
-        data: GraphqlTokenResponse {
-            token: Some(TokenResponse {
-                untracked_volume_usd: volume.to_string(),
-            }),
-        },
-    };
-    Ok(HttpResponse::Ok().json(response))
+    let (platform_id, contract_address) = path.into_inner();
+    if let Some(volume_storage) = storage.get(&platform_id) {
+        let volume = volume_storage.get_volume(&contract_address);
+        let mut contract = ContractSimplified::default();
+        contract.market_data.total_volume.usd = volume.to_f64();
+
+        return Ok(HttpResponse::Ok().json(contract));
+    }
+
+    HttpResponse::BadRequest().message_body(AnyBody::from_message(format!(
+        "Invalid platform_id {}.",
+        &platform_id
+    )))
+}
+
+pub fn config_app(cfg: &mut web::ServiceConfig) {
+    cfg.service(web::resource("/asset_platforms").route(web::get().to(handle_get_asset_platforms)))
```

Comment:
Why do we create many services instead of creating one single service and many routes? What's the reasoning behind it? I'd consider using many routes within the same service,

Reply:
My fault, I didn't realize before that the services are nested.